src/main/java/su/litvak/chromecast/api/v2/Channel.java

```diff
@@ -173,6 +175,18 @@ public void run() {
                     }
                 } catch (InvalidProtocolBufferException ipbe) {
                     LOG.debug("Error while processing protobuf: {}", ipbe.getLocalizedMessage());
+                } catch (JsonParseException jpe) {
+                    LOG.warn("Error while processing protobuf: {}", jpe.getLocalizedMessage());
+                } catch (SocketException se) {
+                    LOG.warn("Socket error, will reconnect: {}", se.getLocalizedMessage());
+                    LOG.debug("StackTrace", se);
+                    try {
+                        close();
```

Comment:
I don't disagree with your design, but I don't see how that would work. Should I introduce a reconnect thread ?